test/martha/martha_v3.test.js

```diff
@@ -424,412 +382,332 @@ test.serial('martha_v3 returns an error when an invalid field is requested', asy
-        '/dg.712C/fc046e84-6cf9-43a3-99cc-ffa2964b88cb',
-    );
-    t.falsy(getJsonFromApiStub.getCall(0).args[1]); // no auth passed
+    t.deepEqual(response.body, bdcDrsMarthaResult(googleSAKeyObject, null));
 
-    const requestedBondSAKeyUrl = getJsonFromApiStub.getCall(1).args[0];
-    const saKeyUrlMatches = requestedBondSAKeyUrl.match(bondSAKeyUrlRegEx);
-    t.truthy(saKeyUrlMatches, 'Bond SA key URL called does not match Bond SA key URL regular expression');
-    const expectedSAKeyProvider = 'fence';
-    const actualSAKeyProvider = saKeyUrlMatches[2];
-    t.is(actualSAKeyProvider, expectedSAKeyProvider);
+    sinon.assert.callCount(getJsonFromApiStub, 2);
 });
 
 test.serial('martha_v3 parses Anvil response correctly', async (t) => {
-    getJsonFromApiStub.onCall(0).resolves(anvilDrsResponse);
-    getJsonFromApiStub.onCall(1).resolves(googleSAKeyObject);
+    const bond = bondUrls('anvil');
+    const { id: objectId, self_uri: drsUri } = anvilDrsResponse;
+    const drs = drsUrls(config.HOST_THE_ANVIL_PROD, objectId);
+    getJsonFromApiStub.withArgs(bond.serviceAccountKeyUrl, terraAuth).resolves(googleSAKeyObject);
+    getJsonFromApiStub.withArgs(drs.objectsUrl, null).resolves(anvilDrsResponse);
     const response = mockResponse();
-    await marthaV3(
-        mockRequest({ body: { 'url': 'drs://dg.ANV0/00008531-03d7-418c-b3d3-b7b22b5381a0' } }),
-        response,
-    );
-    t.is(response.statusCode, 200);
 
-    const result = response.send.lastCall.args[0];
-    sinon.assert.callCount(getJsonFromApiStub, 2); // Bond was called to get SA key
-    t.deepEqual({ ...result }, anvilDrsMarthaResult(googleSAKeyObject, null));
-    t.is(
-        getJsonFromApiStub.getCall(0).args[0],
-        `https://${config.HOST_THE_ANVIL_STAGING}/ga4gh/drs/v1/objects/dg.ANV0/00008531-03d7-418c-b3d3-b7b22b5381a0`,
-    );
-    t.falsy(getJsonFromApiStub.getCall(0).args[1]); // no auth passed
+    await marthaV3(mockRequest({ body: { 'url': drsUri } }), response);
+
+    t.is(response.statusCode, 200);
+    t.deepEqual(response.body, anvilDrsMarthaResult(googleSAKeyObject, null));
 
-    const requestedBondSAKeyUrl = getJsonFromApiStub.getCall(1).args[0];
-    const saKeyUrlMatches = requestedBondSAKeyUrl.match(bondSAKeyUrlRegEx);
-    t.truthy(saKeyUrlMatches, 'Bond SA key URL called does not match Bond SA key URL regular expression');
-    const expectedSAKeyProvider = 'anvil';
-    const actualSAKeyProvider = saKeyUrlMatches[2];
-    t.is(actualSAKeyProvider, expectedSAKeyProvider);
+    sinon.assert.callCount(getJsonFromApiStub, 2);
 });
 
 test.serial('martha_v3 parses a The AnVIL CIB URI response correctly', async (t) => {
-    getJsonFromApiStub.onCall(0).resolves(anvilDrsResponse);
-    getJsonFromApiStub.onCall(1).resolves(googleSAKeyObject);
+    const bond = bondUrls('anvil');
+    const drs = drsUrls(config.HOST_THE_ANVIL_STAGING, 'dg.ANV0%2F00008531-03d7-418c-b3d3-b7b22b5381a0');
```

Comment:
This had me stuck for a bit until I realized I could use `%2F` here instead of `/`. It feels a little arbitrary to me, but I think that's because I don't know all of the history. There's a lot of special case code in the compact identifier resolution, and I would love to write some specific unit tests against that (which, again, would eliminate the need to some of the extra test permutations in this file).